rl4co/utils/meta_trainer.py

```diff
@@ -0,0 +1,245 @@
+from typing import Iterable, List, Optional, Union
+
+import lightning.pytorch as pl
+import torch
+import math
+import copy
+from torch.optim import Adam
+
+from lightning import Callback, Trainer
+from lightning.fabric.accelerators.cuda import num_cuda_devices
+from lightning.pytorch.accelerators import Accelerator
+from lightning.pytorch.core.datamodule import LightningDataModule
+from lightning.pytorch.loggers import Logger
+from lightning.pytorch.strategies import DDPStrategy, Strategy
+from lightning.pytorch.utilities.types import EVAL_DATALOADERS, TRAIN_DATALOADERS
+from rl4co import utils
+import random
+log = utils.get_pylogger(__name__)
+
+
+class MetaModelCallback(Callback):
+    def __init__(self, meta_params, print_log=True):
+        super().__init__()
+        self.meta_params = meta_params
+        assert meta_params["meta_method"] == 'reptile', NotImplementedError
```

Comment:
Another general comment: it seems that this callback is only for Reptile, so we should consider calling it `ReptileCallback`. I think it would be cool to have these as say a `meta_learning/` folder

Reply:
Yes. `ReptileCallback` is better. Where should this new `meta_learning/` folder be located? In `utils/`?

Reply:
I''d say under `models/rl` (which includes `LightningModules`), since `meta_learning` is a way to optimize a policy